File: sample_driver_usages/driver_usage.py
from driver import Lexer

if __name__ == "__main__":
    from pathlib import Path
    import json


    def input_generator():
        for c in "abaaaacdb a":
            yield c


    dfa = json.loads(Path("../states.json").read_text())
    lexer = Lexer(dfa)
    gen = input_generator()
    for i in range(10):
        token = lexer.get_token(gen)
        print(token.token, token.lexeme)

File: sample_driver_usages/driver_usage.js
import fs from "fs";
import path from "path";
import {Lexer} from "./driver.js"; // Adjust to the correct path of your Lexer module

function* inputGenerator() {
    const input = "abaaaacdb a";
    for (const c of input) {
        yield c;
    }
}

// Load DFA from states.json
const dfaPath = path.resolve("states.json");
const dfa = JSON.parse(fs.readFileSync(dfaPath, "utf8"));

// Instantiate the lexer and input generator
const lexer = new Lexer(dfa);
const gen = inputGenerator();

// Run lexer on input
for (let i = 0; i < 10; i++) {
    const token = lexer.getToken(gen);
    console.log(token.token, token.lexeme);
}
